dm/dm/worker/source_worker.go

```diff
@@ -175,7 +175,11 @@ func (w *SourceWorker) Start() {
 				}
 			}
 			if err2 := w.updateSourceStatus(w.ctx); err2 != nil {
-				w.l.Error("failed to update source status", zap.Error(err2))
+				if terror.ErrNoMasterStatus.Equal(err2) {
+					w.l.Warn("This source's bin_log is OFF, so it only supports full_mode.", zap.String("sourceID", w.cfg.SourceID), zap.Error(err2))
```

Comment:
do we need also add this check to pre-check?

Reply:
This may happen before task start and after create source. So we can't pre-check it.

Reply:
But during the pre-check, we also have `binlogChecker`.